lib/galaxy/model/custom_types.py

```diff
@@ -113,149 +113,15 @@ def compare_values(self, x, y):
         return (x == y)
 
 
-class MutationObj(Mutable):
-    """
-    Mutable JSONType for SQLAlchemy from original gist:
-    https://gist.github.com/dbarnett/1730610
-
-    Using minor changes from this fork of the gist:
-    https://gist.github.com/miracle2k/52a031cced285ba9b8cd
-
-    And other minor changes to make it work for us.
-    """
-    @classmethod
-    def coerce(cls, key, value):
-        if isinstance(value, dict) and not isinstance(value, MutationDict):
-            return MutationDict.coerce(key, value)
-        if isinstance(value, list) and not isinstance(value, MutationList):
-            return MutationList.coerce(key, value)
-        return value
-
-    @classmethod
-    def _listen_on_attribute(cls, attribute, coerce, parent_cls):
-        key = attribute.key
-        if parent_cls is not attribute.class_:
-            return
-
-        # rely on "propagate" here
-        parent_cls = attribute.class_
-
-        def load(state, *args):
-            val = state.dict.get(key, None)
-            if coerce and key not in state.unloaded:
-                val = cls.coerce(key, val)
-                state.dict[key] = val
-            if isinstance(val, cls):
-                val._parents[state.obj()] = key
-
-        def set(target, value, oldvalue, initiator):
-            if not isinstance(value, cls):
-                value = cls.coerce(key, value)
-            if isinstance(value, cls):
-                value._parents[target.obj()] = key
-            if isinstance(oldvalue, cls):
-                oldvalue._parents.pop(target.obj(), None)
-            return value
+class JSONType(BaseJSONType):
+    pass
```

Comment:
I'd move this after the definition of `SimpleJSONType` class, so closer to `Mutable.associate_with(JSONType)`.

Also, can you please add a class docstring?